safe_transaction_service/history/models.py

```diff
@@ -1715,11 +1715,23 @@ def created_block_number(self) -> Optional[Type[int]]:
 
 
 class SafeContractDelegateManager(models.Manager):
-    def get_delegates_for_safe(self, address: ChecksumAddress) -> Set[ChecksumAddress]:
-        return set(
-            self.filter(safe_contract_id=address)
-            .values_list("delegate", flat=True)
-            .distinct()
+    def get_delegates_for_safe(
+        self, safe_address: ChecksumAddress
+    ) -> QuerySet["SafeContractDelegate"]:
+        return self.filter(
+            # If safe_contract is null on SafeContractDelegate, delegates are valid for every Safe
```

Comment:
Delegates are valid for every Safe which delegator is an owner. 
This method is returning every delegate where safe_contract is None without consider if the delegator is an owner of that Safe.

Reply:
Fixed